### PR TITLE
White-labeled plugin: Add wpcom-migration key endpoint

### PR DESCRIPTION
**Pull request by sixhours**
## Proposed changes:

* Add a new endpoint to fetch a site's wpcom migration key from the `wpcom-migration` plugin. This is modeled after the migrate guru plugin endpoint but uses different classes and a more accurate endpoint name.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Apply this branch to your AT using Jetpack Beta plugin following instructions in https://github.com/Automattic/jetpack/pull/39377#issuecomment-2346557311
* Apply D161332-code to your sandbox and sandbox `public-api.wordpress.com`. 
* Apply this branch also on your sandbox by running `bin/jetpack-downloader test jetpack-mu-wpcom-plugin add/wpcom-key-endpoint` (make sure this PR and D161332-code are running on your sandbox at the same time)
* Visit the developer console and fetch from `/wpcom/v2/sites/%your-AT-site%/atomic-migration-status/wpcom-migration-key`
* You should see a 403 error
* On the AT site, install and activate the Migrate to WordPress.com plugin (link found in Slack).
* Fetch the endpoint from the developer console; the migration key field should be populated with the same migration key you'll find at `/wp-admin/admin.php?page=wpcom-migration`
* Try to fetch it again and verify that a 403 error is returned

### CODE REVIEW

**Comment by m1r0** [comment]
> Fetch the endpoint from the developer console; the migration key field should be populated

Unfortunately, I wasn't able to fetch the migration key. I'm stuck on getting 403. Here's what I did, hopefully someone can spot my mistake:

* Applied the patch and sandboxed public-api.wordpress.com.
* Applied the branch on the sandbox.
* Created a WoA site (using the WoA Developer Blog Manager tool).
* Installed the Jetpack Beta plugin and activated this branch for the jetpack-mu-wpcom-plugin.
* Added the following to the wp-config.php:
```php
define( 'JETPACK_MU_WPCOM_LOAD_VIA_BETA_PLUGIN', true );
define( 'JETPACK__SANDBOX_DOMAIN', '[my-sandbox-domain]' );
```
* Tested on https://developer.wordpress.com/docs/api/console/

**Comment by m1r0** [comment]
I've also checked if the `wpcom_site_migration_wpcom_migration_key_read` option was created in the database, but it wasn't.

File: projects/packages/jetpack-mu-wpcom/src/features/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-site-migration-wpcom-migration-key.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+		$wpcom_migration_settings = new WPCOMWPSettings();
+		$wpcom_migration_info     = new WPCOMWPInfo( $wpcom_migration_settings );
+
+		update_option( $this->key_is_read_option_name, 'read' );
+
+		return $wpcom_migration_info->getConnectionKey();
+	}
+
+	/**
+	 * Returns Launchpad-related options.
```
Comment (daledupreez) [inline projects/packages/jetpack-mu-wpcom/src/features/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-site-migration-wpcom-migration-key.php]: Nit: This comment seems like a copy-pasta remnant.

**Comment by sixhours** [comment]
@m1r0 That would be user error on my part. I was using the wrong class name for the new plugin so it failed the permissions check. 🙃 That should be resolved with the above updates. Please give it another spin when you get a chance. Thank you!

**Review by m1r0 — APPROVED** [review]
It works now! 🚀

**Pull request merged**